modules/c++/nitf/unittests/test_image_segment_blank_nm_compression.cpp

```diff
@@ -0,0 +1,331 @@
+#include <cstdlib>
+#include <ctime>
+#include <memory>
+#include <iostream>
+#include <iomanip>
+#include <fstream>
+#include <sstream>
+#include <import/nitf.h>
+#include <import/types.h>
+#include <nitf/ImageSegmentComputer.h>
+#include <nitf/ImageSubheader.hpp>
+#include <nitf/Record.hpp>
+#include <nitf/Writer.hpp>
+#include <nitf/Reader.hpp>
+#include <math/Round.h>
+#include <io/FileInputStream.h>
+#include <io/TempFile.h>
+#include "TestCase.h"
+
+extern "C"{
+   NITF_BOOL nitf_ImageIO_getMaskInfo(nitf_ImageIO *nitf,
+                                      nitf_Uint32 *imageDataOffset, nitf_Uint32 *blockRecordLength,
+                                      nitf_Uint32 *padRecordLength, nitf_Uint32 *padPixelValueLength,
+                                      nitf_Uint8 **padValue, nitf_Uint64 **blockMask, nitf_Uint64 **padMask);
+                     
+}
+const int BLOCK_LENGTH = 1024;
+const int64_t ILOC_MAX = 99999;
```

Comment:
For consistency (and to avoid C++11), can you use the NITF typedefs (`nitf::Int64`) instead of the `<ctypes.h>` types?